### DISCUSSION TITLE
Upgrade to 2.4.90 "Soup failed with error 1: Unhandled error"

### DISCUSSION THREAD

**Discussion by alveare1** [Security-Onion-Solutions/securityonion#13434] [2.4] [answered]
### Version

2.4.90

### Installation Method

Security Onion ISO image

### Description

upgrading

### Installation Type

Distributed

### Location

on-prem with Internet access

### Hardware Specs

Exceeds minimum requirements

### CPU

8 to 16 depending on node

### RAM

16 or 32gb

### Storage for /

512

### Storage for /nsm

512

### Network Traffic Collection

other (please provide detail below)

### Network Traffic Speeds

Less than 1Gbps

### Status

No, one or more services are failed (please provide detail below)

### Salt Status

Yes, there are salt failures (please provide detail below)

### Logs

Yes, there are additional clues in /opt/so/log/ (please provide detail below)

### Detail

After doing some troubleshooting so-checkin that fixed the cert issue in nginx. Now it is starting but not auto starting.

Looks like so-mysql and so-soctopus is not starting and the commands for starting them have been removed.

When I manually start ngix, kibana, fleet and soc. The rest of the distributed cluster is all still sitting at 2.4.60

I can manually start logstash, redis, telegraf and idstools. 

logstash starts with a few errors.

----------
          ID: elasticsearch_sbin_jinja
    Function: file.recurse
        Name: /usr/sbin
      Result: False
     Comment: #### /usr/sbin/so-catrust ####
              Failed to cache salt://elasticsearch/tools/sbin_jinja/so-catrust?saltenv=base: File client timed out after 1722837747
              
              #### /usr/sbin/so-elasticsearch-indices-delete-delete ####
              Unable to manage file: File client timed out after 1722837747
     Started: 06:35:49.118619
    Duration: 353161.799 ms
     Changes:   
              ----------
              /usr/sbin:
                  ----------
                  /usr/sbin:
                      ----------
                      user:
                          939
                  user:
                      939
----------
          ID: so-es-cluster-settings
    Function: cmd.run
        Name: /usr/sbin/so-elasticsearch-cluster-settings
      Result: False
     Comment: One or more requisite failed: elasticsearch.config.elasticsearch_sbin_jinja
     Started: 06:42:39.995747
    Duration: 0.012 ms
     Changes:   
----------
          ID: so-elasticsearch-templates
    Function: cmd.run
        Name: /usr/sbin/so-elasticsearch-templates-load
      Result: False
     Comment: One or more requisite failed: elasticsearch.config.elasticsearch_sbin_jinja
     Started: 06:42:40.314884
    Duration: 0.006 ms
     Changes:   
----------
          ID: so-elasticsearch-roles-load
    Function: cmd.run
        Name: /usr/sbin/so-elasticsearch-roles-load
      Result: False
     Comment: One or more requisite failed: elasticsearch.config.elasticsearch_sbin_jinja
     Started: 06:42:40.338686
    Duration: 0.011 ms
     Changes:   

Elastic fleet starts but with this error

----------
          ID: so-elastic-fleet-auto-configure-server-urls
    Function: cmd.run
        Name: /usr/sbin/so-elastic-fleet-urls-update
      Result: False
     Comment: Attempt 1: Returned a result of "False", with the following comment: "Command "/usr/sbin/so-elastic-fleet-urls-update" run"
              Attempt 2: Returned a result of "False", with the following comment: "Command "/usr/sbin/so-elastic-fleet-urls-update" run"
              Attempt 3: Returned a result of "False", with the following comment: "Command "/usr/sbin/so-elastic-fleet-urls-update" run"
              Command "/usr/sbin/so-elastic-fleet-urls-update" run
     Started: 06:19:42.123528
    Duration: 90203.371 ms
     Changes:   

### Guidelines

- [X] I have read the discussion guidelines at https://github.com/Security-Onion-Solutions/securityonion/discussions/1720 and assert that I have followed the guidelines.

**Comment by cm-ops** [comment]
Is there an issue with Elasticsearch? What are the errors in the Logstash log? If this was after soup, would you look in `/root/soup.log` and see if there are any issues?

**Reply by alveare1** [reply]
No issues with Elasticsearch. Issues with Logstash are listed in the original post and one with fleet.

Here is the one time it responded with an "Soup failed with error 1: Unhandled error" if I run soup again it says everything is successful.


[soup.log.2.4.90.20240804.221957.txt](https://github.com/user-attachments/files/16593104/soup.log.2.4.90.20240804.221957.txt)

**Reply by cm-ops** [reply]
What does this return? `sudo salt-call state.show_top`?

This too `sudo salt-call state.apply salt.minion -l info queue=True`

**Reply by alveare1** [reply]
salt-call state.show_top
[INFO    ] Creating module dir '/var/cache/salt/minion/extmods/grains'
[INFO    ] Syncing grains for environment 'base'
[INFO    ] Loading cache from salt://_grains, for base
[INFO    ] Caching directory '_grains/' for environment 'base'
local:
    - The function "state.highstate" is running as PID 773591 and was started at 2024, Aug 15 13:32:10.723075 with jid 20240815133210723075


[salt-call state.apply salt.minion.txt](https://github.com/user-attachments/files/16628308/salt-call.state.apply.salt.minion.txt)

**Comment by cm-ops** [comment]
Thanks, rerun `sudo salt-call state.show_top` looks like a highstate was running when you ran it. If you see the same output:

```
local:
- The function "state.highstate" is running as PID 773591 and was started at 2024, Aug 15 13:32:10.723075 with jid 20240815133210723075
```
kill the PID and rerun.

**Reply by alveare1** [reply]
salt-call state.show_top
[INFO    ] Creating module dir '/var/cache/salt/minion/extmods/grains'
[INFO    ] Syncing grains for environment 'base'
[INFO    ] Loading cache from salt://_grains, for base
[INFO    ] Caching directory '_grains/' for environment 'base'
[INFO    ] Loading fresh modules for state activity
local:
    ----------
    base:
        - ca
        - ssl
        - elasticsearch

**Reply by cm-ops** [reply, marked as answer]
Please try this `sudo cp /opt/so/saltstack/default/salt/top.sls.backup /opt/so/saltstack/default/salt/top.sls`. Then run `sudo so-checkin`.

**Reply by alveare1** [reply]
Wow. What was that I did? Just so I understand. That fixed all of my issue. 

I rebooted after it was done and everything came back up clean. Thank you.

**Reply by cm-ops** [reply]
It appears there was a failure in soup. Glad you are back up and running.
